### PR TITLE
collector: collect process info only when needed

### PR DESCRIPTION
**Pull request by AstroProfundis**
This allows us to collect basic system info and process info of applications separately, e.g., collect system info on host servers, and collect process info for tidb instances.

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -41,19 +41,22 @@ type Metrics struct {
...
 }
 
 func parseOpts() options {
 	optPid := flag.String("pid", "", "The PID of process to collect info. Multiple PIDs can be seperatted by ','.")
+	optProc := flag.Bool("proc", false, "Only collect process info, disabled (Collect everything except process info) by default.")
 	flag.Parse()
 
 	var opts options
 	opts.Pid = *optPid
+	opts.Proc = *optProc
```
Comment (ethercflow) [inline collector/collector.go]: change to
```
opts := options{*optPid, *optPorc}
return &opts
```
maybe better. [(Constructors and composite literals)](https://golang.org/doc/effective_go.html#composite_literals)

File: insight.py [hunk 2]
```diff
@@ -95,13 +95,14 @@ def collector(self, args):
         if args.pid:
             logging.debug(
                 "Collecting process infor only for PID %s" % args.pid)
-            collector_exec = [collector_exec, '-pid', '%s' % args.pid]
+            collector_exec = [collector_exec, '-proc', '-pid', '%s' % args.pid]
```
Comment (ethercflow) [inline insight.py]: How is `-proc`'s value passed in?

File: insight.py [hunk 3]
```diff
@@ -95,13 +95,14 @@ def collector(self, args):
...
         elif args.port:
             protocol = 'UDP' if args.udp else 'TCP'
             pids = ','.join(
                 str(_pid) for _pid in proc_meta.find_process_by_port(args.port, protocol))
             logging.debug("Collecting process infor for PIDs %s" % pids)
-            collector_exec = [collector_exec, '-pid', '%s' % pids]
+            collector_exec = [collector_exec, '-proc', '-pid', '%s' % pids]
```
Comment (ethercflow) [inline insight.py]: Ibid

File: insight.py (hunk 2 above)
Comment (AstroProfundis) [reply insight.py]: `-proc` is a Boolean flag, and no value is needed for it. (default is `false`, set `-proc` makes it `true`, and it's also possible to pass a specific value to it, but that's not necessary)

Comment (ethercflow) [reply insight.py]: Ok, I understand.

**Pull request merged**